course/src/isometry.cc

```diff
@@ -0,0 +1,288 @@
+    r2_ = Vector3(std::initializer_list<double>{*it++, *it++, *it++});
+    r3_ = Vector3(std::initializer_list<double>{*it++, *it++, *it});
+}
+
+Matrix3::Matrix3(const std::initializer_list<double>& l1,
+                 const std::initializer_list<double>& l2,
+                 const std::initializer_list<double>& l3) {
+    r1_ = Vector3(l1);
+    r2_ = Vector3(l2);
+    r3_ = Vector3(l3);
+}
+
+// Operators
+Vector3& Matrix3::operator [] (const int index) {
+    return this->row(index);
+}
+
+const Vector3& Matrix3::operator [] (const int index) const {
+    return this->row(index);
+}
+
+Matrix3 Matrix3::operator + (const Matrix3& m) const {
+    return Matrix3(r1_ + m.r1_, r2_ + m.r2_, r3_ + m.r3_);
+}
+
+Matrix3 Matrix3::operator - (const Matrix3& m) const {
+    return Matrix3(r1_ - m.r1_, r2_ - m.r2_, r3_ - m.r3_);
+}
+
+Matrix3 Matrix3::operator * (const Matrix3& m) const {
+    return Matrix3(r1_ * m.r1_, r2_ * m.r2_, r3_ * m.r3_);
+}
+
+Matrix3 Matrix3::operator * (const double d) const {
+    return Matrix3(r1_ * d, r2_ * d, r3_ * d);
+}
+
+Matrix3 Matrix3::operator / (const Matrix3& m) const {
+    return Matrix3(r1_ / m.r1_, r2_ / m.r2_, r3_ / m.r3_);
+}
+
+Matrix3& Matrix3::operator = (const Matrix3& m) {
+    r1_ = m.r1_;
+    r2_ = m.r2_;
+    r3_ = m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator += (const Matrix3& m) {
+    r1_ += m.r1_;
+    r2_ += m.r2_;
+    r3_ += m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator -= (const Matrix3& m) {
+    r1_ -= m.r1_;
+    r2_ -= m.r2_;
+    r3_ -= m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator *= (const Matrix3& m) {
+    r1_ *= m.r1_;
+    r2_ *= m.r2_;
+    r3_ *= m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator /= (const Matrix3& m) {
+    r1_ /= m.r1_;
+    r2_ /= m.r2_;
+    r3_ /= m.r3_;
+    return *this;
+}
+
+bool Matrix3::operator == (const Matrix3& m) {
+    if ((r1_ == m.r1_) && (r2_ == m.r2_) && (r3_ == m.r3_)) {
+        return true;
+    }
+    return false;
+}
+
+bool Matrix3::operator != (const Matrix3& m) {
+    return !(*this == m);
+}
+
+// Getters
+Vector3& Matrix3::row(int index) {
+    switch(index) {
+        case 0:
+            return r1_;
+        case 1:
+            return r2_;
+        case 2:
+            return r3_;
+        default:
+            throw "Error. Invalid row index for Matrix3";
+    }
+}
+
+const Vector3& Matrix3::row(int index) const {
+    switch(index) {
+        case 0:
+            return r1_;
+        case 1:
+            return r2_;
+        case 2:
+            return r3_;
+        default:
+            throw "Error. Invalid row index for Matrix3";
+    }
+}
+
+const Vector3 Matrix3::col(int index) const {
+    if (index > 2) {
+        throw "Error. Invalid column index for Matrix3";
+    }
+    return Vector3(r1_[index], r2_[index], r3_[index]);
+}
+
+// Computations
+double Matrix3::det() const {
+    double subdet1 =  r1_[0] * (r2_[1] * r3_[2] - r2_[2] * r3_[1]);
```

Comment:
My Algebra is a bit rusty, but I think the subdeterminant is only the thing on the right hand side. I'm being picky. I know.

Reply:
You are completely right. My bad!